### PR TITLE
feat: replace rotate with rekey

### PR DESCRIPTION
**Pull request by ackRow**

### CODE REVIEW

**Review by HatemMn — COMMENTED** [review]
just some ideas. Other than the export * thing, everything is fine I think !

File: src/structs/objects.ts [hunk 1]
```diff
@@ -329,3 +329,73 @@ export class AccessPolicyKms {
     return this.fromAttributes(keyValue.attributes)
   }
 }
+export class RekeyActionKmsBuilder {
```
[the remark this answers is not in the document]
Comment (ackRow) [reply src/structs/objects.ts:332]: After some testing, it seems removing the `export` keyword is causing the `npm pack` to fail

File: src/kms.ts [hunk 2]
```diff
@@ -1181,11 +1161,135 @@ export class KmsClient {
...
+   * Rekeyed User Decryption Keys however will be able to decrypt data encrypted by the previous Master Public Key and
+   * the rekeyed one.
+   * @param {string} privateMasterKeyUniqueIdentifier the unique identifier of the Private Master Key
+   * @param {string} accessPolicy to rekey e.g. "Department::MKG && Security Level::Confidential"
+   * @returns {string[]} returns the updated master keys uids
+   */
+  public async rekeyCoverCryptAccessPolicy(
+    privateMasterKeyUniqueIdentifier: string,
+    accessPolicy: string,
+  ): Promise<string[]> {
```
[the remark this answers is not in the document]
Comment (ackRow) [reply src/kms.ts:1187]: I agree that this is not elegant but the return values are actually public and private key UIDs and not the key object
so we simply return them as String value

File: tests/KMS.test.ts [hunk 3]
```diff
@@ -527,94 +527,189 @@ test(
...
-      `${userKeyID}-HACK`,
-      { bytes: userKey.bytes(), policy: "Security::TopSecret" },
+      `${simpleUser}-HACK`,
+      { bytes: simpleUserKey.bytes(), policy: "Security::TopSecret" },
       {
         link: [new Link(LinkType.ParentLink, mskID)],
       },
     )
-    expect(temperedUserKeyID).toEqual(`${userKeyID}-HACK`)
+    expect(temperedUserKeyID).toEqual(`${simpleUser}-HACK`)
```
[the remark this answers is not in the document]
Comment (ackRow) [reply tests/KMS.test.ts:568]: This PR made some modifications that changes the existing test and also added new methods to test
We can try to split the test in 2

File: tests/KMS.test.ts [hunk 4]
```diff
@@ -527,94 +527,189 @@ test(
...
+
+    // Top secret user can read the ciphertext
+    {
```
[the remark this answers is not in the document]
Comment (ackRow) [reply tests/KMS.test.ts:571]: Exactly :+1:

**Pull request merged**